spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/util/LazyUtil.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2017-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.datastore.core.util;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.util.List;
+import java.util.function.Supplier;
+
+import org.springframework.cloud.gcp.data.datastore.core.mapping.DatastoreDataException;
+import org.springframework.util.Assert;
+
+/**
+ * @author Dmitry Solomakha
+ */
+final public class LazyUtil {
+
+	private LazyUtil() {
+	}
+
+	public static <T> T wrapSimpleLazyProxy(Supplier<T> supplierFunc, Class<T> type, List keys) {
+		return (T) Proxy.newProxyInstance(type.getClassLoader(), new Class[] {type},
+				new SimpleLazyDynamicInvocationHandler<T>(supplierFunc, keys));
+	}
+
+	public static boolean hasUsableKeys(Object object) {
```

Comment:
For the public methods of this class, can you add a javadoc explaining how the methods should be used?